File: sol/3033.py
class Solution:
    def modifiedMatrix(self, matrix: List[List[int]]) -> List[List[int]]:
        m = len(matrix)
        n = len(matrix[0])
        colMax = tuple(max(c) for c in zip(*matrix))
        for i in range(m):
            for j in range(n):
                if matrix[i][j] == -1:
                    matrix[i][j] = colMax[j]
        return matrix

File: sol/3033/3033.cpp
struct Solution {
    vector<vector<int>> modifiedMatrix(vector<vector<int>>& matrix) {
        int m = matrix.size(), n = matrix[0].size();
        for (int j = 0; j < n; ++j){
            vector<int> idx(0);
            int maxi = -1;
            for (int i = 0; i < m; ++i){
                maxi = max(maxi, matrix[i][j]);
                if (matrix[i][j] == -1){
                    idx.push_back(i);
                }
            }
            for (int i : idx){
                matrix[i][j] = maxi;
            }
        }
        return matrix;
    }
};
